server/src/main/java/org/opensearch/index/shard/IndexShard.java

```diff
@@ -3100,6 +3104,12 @@ private EngineConfig newEngineConfig(LongSupplier globalCheckpointSupplier) {
             }
         };
 
+        final List<ReferenceManager.RefreshListener> internalRefreshListener;
+        if (remoteStoreRefreshListener != null && shardRouting.primary()) {
+            internalRefreshListener = Arrays.asList(new RefreshMetricUpdater(refreshMetric), remoteStoreRefreshListener);
+        } else {
+            internalRefreshListener = Collections.singletonList(new RefreshMetricUpdater(refreshMetric));
+        }
```

Comment:
Lets push this down to `IndexShard` and pass a `NoopRemoteStoreRefreshListener` as needed?

Reply:
Sorry I meant IndexServices that create `IndexShard `

Reply:
I am fine with what we have now, but if the checks grow at multiple place we might want to give this a revisit

Reply:
This is identical to the listener for segrep - [here](https://github.com/opensearch-project/OpenSearch/pull/3108#discussion_r870624373).  What about building a list of additional listeners in IndexService and pass directly to IndexShard?

Reply:
@Bukhtawar Agree, adding a task for the same.
@mch2 Yes, once we merge with SegRep code, we can abstract it out.